src/main/java/software/amazon/awssdk/crt/http/HttpStream.java

```diff
@@ -55,6 +63,57 @@ public void incrementWindow(int windowSize) {
         }
     }
 
+    /**
+     * Use only for Http 1.1 Chunked Encoding. At some later point we may adapt this interface for H2, but not yet. If
+     * using chunked encoding, remember, you'll need to send an empty chunk to signal the end of your stream. You must call
+     * activate() before using this function.
+     *
+     * @param chunkData chunk of data to send.
+     * @param chunkCompletionCallback Invoked upon the data being flushed to the wire or an error occurring.
+     */
+    public void writeChunk(final byte[] chunkData, final HttpStreamWriteChunkCompletionCallback chunkCompletionCallback) {
+        if (chunkCompletionCallback == null) {
+            throw new IllegalArgumentException("You must supply a chunkCompletionCallback");
+        }
+
+        if (chunkData == null) {
+            throw new IllegalArgumentException("You must provide a non-null chunkData");
+        }
+
+        int error = httpStreamWriteChunk(getNativeHandle(), chunkData, chunkCompletionCallback);
+
+        if (error != 0) {
+            int lastError = CRT.awsLastError();
+            throw new CrtRuntimeException(lastError);
+        }
+    }
+
+    /**
+     * Use only for Http 1.1 Chunked Encoding. At some later point we may adapt this interface for H2, but not yet. If
+     * using chunked encoding, remember, you'll need to send an empty chunk to signal the end of your stream. You must call
+     * activate() before using this function.
+     *
+     * @param chunkData chunk of data to send.
+     * @return completable future which will complete upon the data being flushed to the wire or an error occurring.
+     */
+    public CompletableFuture<Void> writeChunk(final byte[] chunkData) {
```

Comment:
Should we make a `finishChunks()` or something that automates writing an empty chunk to finish the stream?